code/src/layouts/header/hooks/use-sticky-header/useStickyHeader.test.ts

```diff
@@ -0,0 +1,141 @@
+import { renderHook } from "@testing-library/react";
+import useStickyHeader from "./useStickyHeader";
+
+type RenderUseStickyHeaderOptions = {
+  headerElement?: HTMLDivElement | null;
+  scrollHandleElement?: HTMLDivElement | null;
+};
+
+const mockedObserveFn = jest.fn();
+const mockedUnobserveFn = jest.fn();
+
+let observerOptions: Partial<IntersectionObserverInit> | undefined;
+let observerCallback: (entry?: { isIntersecting: boolean }) => void;
+
+const rightIntersectionOptions = { rootMargin: "300px" };
+const headerStyleWithoutIntersection = "--top: -112px";
+const headerStyleWithIntersection = "--top: 0";
+
+global.IntersectionObserver = jest.fn((callback, options) => {
+  observerCallback = (entry = { isIntersecting: true }) => {
+    callback([entry as IntersectionObserverEntry], {} as IntersectionObserver);
+  };
+
+  observerOptions = options;
+
+  return {
+    observe: mockedObserveFn,
+    unobserve: mockedUnobserveFn,
+    disconnect: jest.fn(),
+    takeRecords: jest.fn(),
+    root: null,
+    rootMargin: "0px",
+    thresholds: []
+  };
+});
+
+let initialHeaderElement = document.createElement("div");
+let initialScrollHandleElement = document.createElement("div");
+
+const defaultOptions = {
+  headerElement: initialHeaderElement,
+  scrollHandleElement: initialScrollHandleElement
+};
+
+const renderUseStickyHeader = ({
+  headerElement = initialHeaderElement,
+  scrollHandleElement = initialScrollHandleElement
+}: RenderUseStickyHeaderOptions = defaultOptions) => {
+  const returnedValue = renderHook(() => {
+    const hookReturnedValue = useStickyHeader();
+
+    headerElement &&
+      Object.defineProperty(headerElement, "clientHeight", {
+        configurable: true,
+        value: 112
+      });
+
+    hookReturnedValue.headerRef.current = headerElement;
+    hookReturnedValue.scrollHandleRef.current = scrollHandleElement;
+
+    return hookReturnedValue;
+  });
+
+  return returnedValue;
+};
+
+describe("Test useStickyHeader", () => {
+  afterEach(() => {
+    initialHeaderElement = document.createElement("div");
+    initialScrollHandleElement = document.createElement("div");
+
+    observerOptions = undefined;
+    observerCallback = () => {};
+
+    jest.clearAllMocks();
+  });
+
+  it("Should not call observe function or return ref if there is no at least one ref assigned to html element", async () => {
```

Comment:
can we avoid using `async` for each case?

Reply:
Thanks for noticing, I mistakenly added it. FIxed